backend/coreapp/asm_diff_wrapper.py

```diff
@@ -120,38 +124,34 @@ def diff(target_assembly: Assembly, platform: str, diff_label:Optional[str], com
             logger.info("Base asm empty - attempting to regenerate")
             compiler_wrapper.CompilerWrapper.assemble_asm(arch.name, target_assembly.source_asm, target_assembly)
             if len(target_assembly.elf_object) == 0:
-                logger.error("Regeneration of base-asm failed")
-                return {"error": "Error: Base asm empty"}
+                raise AssemblyError("Regeneration of base-asm failed: base asm empty")
 
         basedump = AsmDifferWrapper.run_objdump(target_assembly.elf_object, platform, config, diff_label)
         if not basedump:
-            return {"error": "Error running asm-differ on basedump"}
+            raise ObjdumpError("Error running objdump on base")
 
         if len(compiled_elf) == 0:
-            logger.error("Creation of compilation elf_object failed")
-            return {"error": "Error: Compialtion elf_object failed"}
+            raise CompilationError("Compilation of code for diff failed")
 
         mydump = AsmDifferWrapper.run_objdump(compiled_elf, platform, config, diff_label)
         if not mydump:
-            return {"error": "Error running asm-differ on mydump"}
+            raise ObjdumpError("Error running objdump on new")
 
         # Preprocess the dumps
         basedump = asm_differ.preprocess_objdump_out(None, target_assembly.elf_object, basedump)
         mydump = asm_differ.preprocess_objdump_out(None, compiled_elf, mydump)
 
         try:
             display = asm_differ.Display(basedump, mydump, config)
-        except Exception:
-            logger.exception("Error running asm-differ")
-            return {"error": "Error running asm-differ"}
+        except Exception as e:
+            raise DiffError(f"Error running asm-differ: {str(e)}")
 
         try:
             # TODO: It would be nice to get a python object from `run_diff()` to avoid the
             # JSON roundtrip. See https://github.com/simonlindholm/asm-differ/issues/56
             result = json.loads(display.run_diff()[0])
             result["error"] = None
-        except Exception:
-            logger.exception("Error running asm-differ")
-            return {"error": "Error running asm-differ"}
+        except Exception as e:
+            raise DiffError(f"Error running asm-differ: {str(e)}")
```

Comment:
What happens without the str() for these exceptions? Is string representation the default?

Reply:
I'm not sure...this is what was suggested to me by copilot. @zbanks do you know if this is preferred / safe, or it's okay to just use e directly?

Reply:
It defaults to using the [`__format__` method](https://docs.python.org/3/library/functions.html#format) on the object, instead of `__str__`, though I believe the default implementation of that method should be equivalent to `str(...)` for these exceptions. 
I think `{e}` would be preferred here?

Reply:
great, thanks!